crates/jstz_engine/src/context.rs

```diff
@@ -101,4 +164,25 @@ mod test {
 
         assert_eq!(raw_cx, unsafe { cx.as_raw_ptr() })
     }
+
+    #[test]
+    fn entering_and_leaving_realm() {
+        let engine = mozjs::rust::JSEngine::init().unwrap();
+        let rt = Runtime::new(engine.handle());
+        let cx = &mut Context::from_runtime(&rt);
+
+        // Enter a new realm to evaluate the script in.
+        let mut cx1 = cx.new_realm().unwrap();
+        let global1 = unsafe { JS::CurrentGlobalOrNull(cx1.as_raw_ptr()) };
+
+        let cx2 = cx1.new_realm().unwrap();
+        let global2 = unsafe { JS::CurrentGlobalOrNull(cx2.as_raw_ptr()) };
+        assert_ne!(global1, global2);
+
+        drop(cx2);
+
+        // Dropping the entered realm should restore the previous realm
+        let global3 = unsafe { JS::CurrentGlobalOrNull(cx1.as_raw_ptr()) };
+        assert_eq!(global1, global3);
```

Comment:
What do you think of testing against the context pointer to show that the real does indeed get pushed on and popped off? 

```suggestion
        let ptr = unsafe { cx1.as_raw_ptr() };
        let global1 = unsafe { JS::CurrentGlobalOrNull(cx1.as_raw_ptr()) };
        assert_eq!(global1, unsafe { JS::CurrentGlobalOrNull(ptr) });

        let cx2 = cx1.new_realm().unwrap();
        let global2 = unsafe { JS::CurrentGlobalOrNull(cx2.as_raw_ptr()) };
        assert_ne!(global1, global2);
        assert_eq!(global2, unsafe { JS::CurrentGlobalOrNull(ptr) });

        drop(cx2);

        assert_eq!(global1, unsafe { JS::CurrentGlobalOrNull(ptr) });
```

Reply:
Sure!